bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -745,6 +748,19 @@ private static boolean checkIsStreamType(BType sourceType, BStreamType targetTyp
                                unresolvedTypes);
     }
 
+    private static boolean checkIsTableType(BType sourceType, BTableType targetType, List<TypePair> unresolvedTypes) {
+        if (sourceType.getTag() != TypeTags.TABLE_TAG) {
+            return false;
+        }
+        boolean isTableType = checkContraints(((BTableType) sourceType).getConstrainedType(),
+                targetType.getConstrainedType(), unresolvedTypes);
+        if (isTableType && ((BTableType) sourceType).getKeyType() != null) {
```

Comment:
in runtime, we have to check the key specifiers, not the key type @pcnfernando

Reply:
@gimantha